frontend/components/SetupPage/AgentIntroduction/AgentIntroduction.tsx

```diff
@@ -0,0 +1,143 @@
+import { Divider, Flex, Typography } from 'antd';
+import React, { FC, useCallback, useMemo, useState } from 'react';
+
+import { SetupScreen } from '@/enums/SetupScreen';
+import { useServices } from '@/hooks/useServices';
+import { useSetup } from '@/hooks/useSetup';
+
+import { IntroductionStep, OnboardingStep } from './IntroductionStep';
+
+const { Text } = Typography;
+
+type IntroductionProps = {
+  steps: OnboardingStep[];
+  onOnboardingComplete: () => void;
+};
+
+const Introduction = ({ steps, onOnboardingComplete }: IntroductionProps) => {
+  const { goto } = useSetup();
+  const [currentStep, setCurrentStep] = useState(0);
+
+  const onNextStep = useCallback(() => {
+    if (currentStep === steps.length - 1) {
+      onOnboardingComplete();
+    } else {
+      setCurrentStep((prev) => (prev + 1) % steps.length);
+    }
+  }, [currentStep, steps.length, onOnboardingComplete]);
+
+  const onPreviousStep = useCallback(() => {
+    if (currentStep === 0) {
+      goto(SetupScreen.AgentSelection);
+    } else {
+      setCurrentStep((prev) => prev - 1);
+    }
+  }, [currentStep, goto]);
+
+  return (
+    <IntroductionStep
+      title={steps[currentStep].title}
+      desc={steps[currentStep].desc}
+      imgSrc={steps[currentStep].imgSrc}
+      helper={steps[currentStep].helper}
+      btnText={currentStep === steps.length - 1 ? 'Set up agent' : 'Continue'}
+      onPrev={onPreviousStep}
+      onNext={onNextStep}
+    />
+  );
+};
+
+const predictionAgentSteps: OnboardingStep[] = [
```

Comment:
maybe we can move it to `frontend/components/SetupPage/AgentIntroduction/constants.ts` or even `frontend/components/SetupPage/AgentIntroduction/steps.ts` would be good to separate in case there will be more?